File: user/src/controllers/comment.controller.js
import commentService from "../services/comment.service.js"

const getAllComments = async (req, res) => {
    try {
        const comments = await commentService.getAllComments();
        res.status(200).json(comments);
    } catch (error) {
        res.status(500).json({ message: error.message });
    }
};

const getCommentByCommenterId = async (req, res) => {
    const commenterId = req.params.id; 
    try {
        const commenter = await commentService.getCommentByCommenterId(commenterId);
        res.status(200).json(commenter); 
    } catch (error) {
        res.status(500).json({ message: error.message });
    }
};

const getCommentBySuggestionId = async (req, res) => {
    const suggestionId = req.params.id; 
    try {
        const suggestion = await commentService.getCommentBySuggestionId(suggestionId);
        res.status(200).json(suggestion);
    } catch (error) {
        res.status(500).json({ message: error.message });
    }
};

const createComment = async (req, res) => {
    try {
        const comment = await commentService.createComment(req.body);
        res.status(201).json(comment);
    } catch (error) {
        res.status(400).json({ message: error.message });
    }
};

const deleteComment = async (req, res) => {
    const id = req.params.id;
    try {
        await commentService.deleteComment(id);
        res.status(204).send(); 
    } catch (error) {
        res.status(400).json({ message: error.message });
    }
};

export { getAllComments, getCommentByCommenterId, getCommentBySuggestionId, createComment, deleteComment };

File: Admin-Spring/src/main/java/com/ibm/springboot/cityManagement/controller/CommentController.java
//package com.ibm.springboot.cityManagement.controller;
//
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.http.ResponseEntity;
//import org.springframework.web.bind.annotation.*;
//
//import com.ibm.springboot.cityManagement.model.Comment;
//import com.ibm.springboot.cityManagement.service.CommentService;
//
//@RestController
//@RequestMapping("/comments")
//public class CommentController {
//
//    @Autowired
//    private CommentService commentService;
//
//    @GetMapping("/comments")
//    public ResponseEntity<?> getAllComments() {
//        try {
//            return ResponseEntity.ok(commentService.getAllComments());
//        } catch (Exception e) {
//            return ResponseEntity.status(500).body(e.getMessage());
//        }
//    }
//
//    @GetMapping("/commenter/{commenterId}")
//    public ResponseEntity<?> getCommentByCommenterId(@PathVariable String commenterId) {
//        try {
//            return ResponseEntity.ok(commentService.getCommentByCommenterId(commenterId));
//        } catch (Exception e) {
//            return ResponseEntity.status(500).body(e.getMessage());
//        }
//    }
//
//    @GetMapping("/suggestion/{suggestionId}")
//    public ResponseEntity<?> getCommentBySuggestionId(@PathVariable String suggestionId) {
//        try {
//            return ResponseEntity.ok(commentService.getSuggestionBySuggestionId(suggestionId));
//        } catch (Exception e) {
//            return ResponseEntity.status(500).body(e.getMessage());
//        }
//    }
//
//    @PostMapping("/comments")
//    public ResponseEntity<?> createComment(@RequestBody Comment comment) {
//        try {
//            return ResponseEntity.status(201).body(commentService.createComment(comment));
//        } catch (Exception e) {
//            return ResponseEntity.status(400).body(e.getMessage());
//        }
//    }
//
//    @DeleteMapping("/{id}")
//    public ResponseEntity<?> deleteComment(@PathVariable String id) {
//        try {
//            commentService.deleteComment(id);
//            return ResponseEntity.status(204).build();
//        } catch (Exception e) {
//            return ResponseEntity.status(400).body(e.getMessage());
//        }
//    }
//}

